src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/wwwroot/BlazorWeb-CSharp.lib.module.js

```diff
@@ -0,0 +1,77 @@
+async function fetchWithErrorHandling(url, options = {}) {
+    const response = await fetch(url, {
+        credentials: 'include',
+        ...options
+    });
+    if (!response.ok) {
+        const text = await response.text();
+        console.error(text);
+        throw new Error(`The server responded with status ${response.status}.`);
+    }
+    return response;
+}
+
+async function createCredential() {
+    const optionsResponse = await fetchWithErrorHandling('/Account/PasskeyCreationOptions', {
+        method: 'POST',
+    });
+    const optionsJson = await optionsResponse.json();
+    const options = PublicKeyCredential.parseCreationOptionsFromJSON(optionsJson);
+    return await navigator.credentials.create({ publicKey: options });
+}
+
+async function requestCredential(email) {
+    const optionsResponse = await fetchWithErrorHandling(`/Account/PasskeyRequestOptions?username=${email}`, {
+        method: 'POST',
+    });
+    const optionsJson = await optionsResponse.json();
+    const options = PublicKeyCredential.parseRequestOptionsFromJSON(optionsJson);
+    return await navigator.credentials.get({ publicKey: options });
+}
+
+customElements.define('passkey-submit', class extends HTMLElement {
```

Comment:
Have you considered https://web.dev/articles/more-capable-form-controls it seems on the ballpark of a bunch of work you are doing here.

Reply:
I adjusted the approach to use a form-associated custom element. It was a small code change, but I think it's an improvement as we don't need to programmatically add hidden input elements to the DOM anymore.